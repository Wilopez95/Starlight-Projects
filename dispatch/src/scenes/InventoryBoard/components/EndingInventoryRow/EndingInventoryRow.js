/* eslint-disable react/prop-types */

import get from 'lodash/get';
import styled from 'styled-components';
import { device } from '@root/utils/device';

const TableRow = styled.tr`
  text-align: center !important;
  border-bottom: 1px solid #f0f2f3 !important;
  border-right: 1px solid #d1d3d4 !important;
`;
const TitleSmall = styled.td`
  background-color: #434343 !important;
  color: white !important;
  text-align: right !important;
  font-weight: 14px !important;
  border-right: 1px solid #434343 !important;
  border-bottom: 1px solid #f0f2f3 !important;
  font-size: 69% !important;

  @media ${device.laptop} {
    background-color: #434343 !important;
    color: white !important;
    text-align: right !important;
    border-right: 1px solid #434343 !important;
    font-size: 0.98em !important;
    width: 100% !important;
    word-wrap: none;
    font-weight: 700 !important;
    margin-left: -5px !important;
    margin-right: 10px;
    white-space: nowrap;
  }
`;

const TotalsTd = styled.td`
  text-align: center !important;
  border-right: 1px solid #d1d3d4 !important;
  font-size: 1.5em !important;
  font-weight: 700 !important;
`;

// type Props = {
//   totalPerSizeMapping: Object,
//   size: string,
//   // filtered ending inventory values for today
//   endDayOne: number,
//   // filtered ending inventory values for tomorrow
//   endDayTwo: number,
//   // filtered ending inventory values for day 3
//   endDayThree: number,
//   // filtered ending inventory values for day 4
//   endDayFour: number,
//   finalDayOne: number,
//   finalDayTwo: number,
//   finalDayThree: number,
//   finalDayFour: number,
// };

function EndingInventoryRow(props) {
  // get table data
  const {
    size,
    totalPerSizeMapping,
    endDayOne,
    endDayTwo,
    endDayThree,
    endDayFour,
    finalDayOne,
    finalDayTwo,
    finalDayThree,
    finalDayFour,
  } = props;

  // Ending inventory = + 1 for Final and - 1 for delivery (SPOT)
  return (
    <TableRow>
      <TitleSmall>Ending Inventory</TitleSmall>
      <TotalsTd>{get(totalPerSizeMapping, size, 0) - endDayOne + finalDayOne || 0}</TotalsTd>
      <TotalsTd>{get(totalPerSizeMapping, size, 0) - endDayTwo + finalDayTwo || 0}</TotalsTd>
      <TotalsTd>{get(totalPerSizeMapping, size, 0) - endDayThree + finalDayThree || 0}</TotalsTd>
      <TotalsTd>{get(totalPerSizeMapping, size, 0) - endDayFour + finalDayFour || 0}</TotalsTd>
    </TableRow>
  );
}

export default EndingInventoryRow;
