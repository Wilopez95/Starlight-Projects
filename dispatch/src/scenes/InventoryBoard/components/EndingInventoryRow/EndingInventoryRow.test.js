import { shallow } from 'enzyme';
// import sinon from 'sinon';
import {
  CURRENT_INVENTORY,
  FINALS,
  // WORK_ORDERS,
} from 'scenes/InventoryBoard/__fixtures__/inventoryboard.fixture';
import EndingInventoryRow from './EndingInventoryRow';

// for tests to work with react media
global.matchMedia = media => ({
  addListener: () => {},
  removeListener: () => {},
  matches: media === '(min-width: 545px)',
});
describe('<EndingInventoryRow />', () => {
  const defaultProps = {
    totalPerSizeMapping: {
      10: 43,
      12: 43,
      20: 27,
      30: 91,
      40: 22,
      '40CT': 38,
    },
    size: '20',
    tallyOnTruckSizes: {
      TRUCK: {
        10: 2,
        12: 4,
        30: 15,
        40: 3,
        '40CT': 2,
      },
    },
    finals: FINALS,
    endingInventory: CURRENT_INVENTORY,
    today: '2019-09-04',
    tomorrow: '2019-09-05',
    thirdDay: '2019-09-06',
    fourthDay: '2019-09-07',
  };

  it('should be successfully mounted', () => {
    const wrapper = shallow(<EndingInventoryRow {...defaultProps} />);
    expect(wrapper).toBeDefined();
  });
  it('should  match the snapshot', () => {
    const wrapper = shallow(<EndingInventoryRow {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
