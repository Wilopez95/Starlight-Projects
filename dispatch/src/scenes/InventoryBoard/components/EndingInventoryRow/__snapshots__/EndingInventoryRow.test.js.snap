// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EndingInventoryRow /> should  match the snapshot 1`] = `ShallowWrapper {}`;
